schema/data/fixtures/login-redirects-setup.sql

```diff
@@ -0,0 +1,145 @@
+('[
+  {
+    "fuelRowId": 13,
+    "quantity": 4,
+    "fuelUnits": "t",
+    "emissionCategoryRowId": 1
+  },
+  {
+    "fuelRowId": 11,
+    "quantity": 400,
+    "fuelUnits": "kL",
+    "emissionCategoryRowId": 2
+  }
+]'),
+('[
+  {
+    "productAmount": 8760000,
+    "productRowId": 29,
+    "productUnits": "MWh",
+    "productEmissions": 5900,
+    "requiresEmissionAllocation": true,
+    "requiresProductAmount": true,
+    "isCiipProduct": true
+  },
+  {
+    "productAmount": 12000,
+    "productRowId": 26,
+    "productUnits": "t",
+    "productEmissions": 1300,
+    "requiresEmissionAllocation": true,
+    "requiresProductAmount": true,
+    "isCiipProduct": true
+  }
+]');
+
+create or replace function e2e_test_initialize_form(
+  form_id_start int,
+  application_id_input int
+)
+returns boolean as
+$function$
+  declare
+    temp_row record;
+    loop_counter int default 1;
+  begin
+    -- Populate initial form results for each form section:
+    for temp_row in
+      select row_number() over () as index, form_id from ggircs_portal.ciip_application_wizard
+    loop
+      insert into ggircs_portal.form_result(id, form_id, application_id, version_number, form_result, created_by)
+        overriding system value
+        values (form_id_start + loop_counter, temp_row.form_id, application_id_input, 1, (select dummy_result from dummy_results where id = temp_row.index), 6);
+      insert into ggircs_portal.form_result_status(application_id, form_id, form_result_status)
+      values (application_id_input, temp_row.form_id, 'in review');
+
+      loop_counter := loop_counter + 1;
+    end loop;
+  return 'true';
+end;
+$function$ language plpgsql strict volatile;
+grant execute on function e2e_test_initialize_form to ciip_administrator, ciip_analyst, ciip_industry_user;
+
+-- Initializes form results
+select e2e_test_initialize_form(200, 101);
+select e2e_test_initialize_form(300, 102);
+select e2e_test_initialize_form(400, 103);
```

Comment:
why is all of this needed? can't you set a reporting year and use `create_application_mutation_chain` instead?

Reply:
What's necessary is to be able to explicitly set ids (overriding id generation) in order for the test to have known ids to compute the URL for certain routes with required query params, eg: `/reporter/application?applicationId=blah&version=1&formResultId=blah`. 

Unfortunately that seems to mean copying a bunch of the things that `create_application_mutation_chain` and `create_application_revision_mutation_chain` do. This is what @dleard and I were working on yesterday.

Similar affected routes are listed in REQUIRED_QUERY_PARAMS in the login-redirects test.

Reply:
if you do `truncate cascade restart identity` on the application table at the beginning of the fixture, you should have known ids without needing to override id generation

Reply:
Thanks! Have updated accordingly